tests/integration/it_poster_test.py

```diff
@@ -1,15 +1,30 @@
-import pytest
+from unittest import mock
 
-from timeless.poster.api import Poster, Authenticated
+from timeless.poster.api import Authenticated, PosterAuthData
 
 """Integration tests for Poster"""
 
 
-@pytest.mark.skip(
-    reason="Authentication mechanism (Authenticated#auth()) is not yet implemented!"
-)
-def test_auth():
-    assert Authenticated(
-        clientid="$0m3C1i3ntId"
-    ).access_token(), "Poster did not authenticate the user!"
+@mock.patch("timeless.poster.api.requests")
+def test_auth(requests_mock):
+    auth_data = PosterAuthData(
+        application_id="test_id",
+        application_secret="test_secret",
+        redirect_uri="test_uri",
+        code="test_code",
+    )
+    auth_token = "861052:02391570ff9af128e93c5a771055ba88"
+    json_object = mock.Mock()
+    json_object.return_value = {"access_token": auth_token}
 
+    requests_mock.post.return_value = type(
+        'Mock date',
+        (),
+        {
+            'ok': True,
+            'json': json_object,
+        }
+    )
+
+    auth_token = Authenticated(auth_data).auth()
+    assert auth_token == auth_token
```

Comment:
@vladarefiev what do you think about these:
```python
class Response:
    auth_token = "861052:02391570ff9af128e93c5a771055ba88"

    def ok(self):
        return True

    def json(self):
        return {"access_token": self.auth_token}

requests_mock.post.return_value = Response()
assert Authenticated(auth_data).auth() == Response.auth_token
```
it's clearer, I think.

Reply:
@vryazanov like